README.md

```diff
@@ -0,0 +1,134 @@
+- 🟥 `POST: /api/books` - **Add New Book**: Admin users can add new books to the catalog.
+- 🟥 `PUT: /api/books/{id}` - **Update Book Details**: Admin users can update existing book details.
+- 🟥 `DELETE: /api/books/{id}` - **Delete Book**: Admin users can remove books from the catalog.
+
+### Category Management
+
+- 🟨 `GET: /api/categories` - **Retrieve Categories**: Users can view the list of available categories.
+- 🟨 `GET: /api/categories/{id}` - **View Category**: Users can also check information about each category individually.
+- 🟨 `GET: /api/categories/{id}/books` - **Get Books by Category**: Users can retrieve all books belonging to a specific category.
+- 🟥 `POST: /api/categories` - **Create Category**: Admin users can create new categories for books.
+- 🟥 `PUT: /api/categories/{id}` - **Update Category**: Admin users can update category details.
+- 🟥 `DELETE: /api/categories/{id}` - **Delete Category**: Admin users can delete categories.
+
+### Shopping Cart
+
+- 🟨 `GET: /api/cart` - **View Cart**: Users can view the contents of their shopping cart.
+- 🟨 `POST: /api/cart` - **Add to Cart**: Users can add books to their shopping cart.
+- 🟨 `PUT: /api/cart/cart-items/{id}` - **Update Cart**: Users can update the quantity of a specific item in their cart.
+- 🟨 `DELETE: /api/cart/cart-items/{id}` - **Remove from Cart**: Users can remove a specific item from their cart.
+
+### Order Management
+
+- 🟨 `GET: /api/orders` - **View Orders History**: Users can view all their orders history.
+- 🟨 `GET: /api/orders/{id}/items` - **View Order Items**: Users can view all items included in a specific order.
+- 🟨 `GET: /api/orders/{orderId}/items/{itemId}` - **View Specific Order Item**: Users can check details of a particular item within an order.
+- 🟨 `POST: /api/orders` - **Place Order**: Users can place orders for the books in their cart.
+- 🟥 `PATCH: /api/orders/{id}` - **Update Order Status**: Admin users can update the status of orders.
+
+## Video Presentation
+
+For a visual demonstration of how the Bookstore API works and its various functionalities, you can watch the video presentation available [here](https://www.loom.com/share/2464c0541478423c9b88868330317bd8). In this presentation, I walk through the key features of the application, demonstrate its usage with examples, and provide insights into the development process.
+
+## SQL Database Diagram
+Below is a simplified representation of the database schema used in the Bookstore API:
+
+![bookstore-db-diagram](bookstore-db-diagram.png)
+
+This diagram illustrates the relationships between different entities in the database, including tables for users, books, categories, shopping carts, orders, and more.
+
+## Setup Instructions
+
+1. Clone the repository to your local machine.
+2. Ensure you have Docker installed and Docker Engine running.
+3. Navigate to the project directory.
+4. Run `./mvnw clean package` to create a .jar file.
```

Comment:
I have a proposition for you regarding the steps below to split the instruction into two different approaches:
a) run with Maven
b) run with Docker Compose

1. Copy and modify current Compose file to feature only database configuration. Name the old copy "docker-compose-full.yml" and the new one "docker-compose.yml".
2. Run app with Spring Boot Maven plugin and docker compose dependency which handles db startup on its own.
3. Provide two separate instructions for both strategies:
- db-only should end with 3 steps: clone, navigate to dir and run spring boot with mvn command
- full Compose - more complex, with Compose command to be execuded with `-f`flag at the beginning, referring to desired Compose file.